MiniProjectOne/Main.py

```diff
@@ -282,13 +305,149 @@ def process(self, args):
+                            AND p.date < dt.now()
+                            """
+                        )
+
+            # accept payment if amountPaid + payment Amount <= fine, otherwise raise error
+            totalAmount = amountPaid + amount
+
+            if fine >= totalAmount:
+                # Update fine amount for the ticket
+                fineRemaining = fine - totalAmount
+                self.c.execute(
+                    "UPDATE tickets SET fine = ? WHERE tno = ?",
+                    (fineRemaining, tno)
+                )
+                self.conn.commit()
+
+                # Insert amount into payments table
+                self.c.execute(
+                    """
+                    INSERT INTO payments (tno, pdate, amount) 
+                    VALUES (?,?,?)
+                    """,
+                    (tno, pdate, amount)
+                )
+                self.conn.commit()
+            else:
+                raise Exception("Amount paid exceeds ticket fine")
         else:
-            # TODO: Process a payment
-            pass
+            raise Exception("Missing Argument(s)")
 
     def getAbstract(self, args):
-        """
-        Return a drivers abstract
+        # get the person's name from the user
+        fname = input("Enter first name of the person: ")
+        lname = input("Enter last name of the person: ")
+
+        # Ensure the person belongs to database
+        person = self.c.execute(
+            """
+            SELECT regno FROM births b1, births b2 WHERE b1.regno = b2.regno
+            AND b1.fname = ? and b2.lname = ?
+            """,
+            (fname, lname)
+        )
+        if person == None:
+            print("Invalid person entry")
+        
+        # get driver's abstract
+        # ToDo: finish the query
+
+        # sort tickets
+        count = 0
+        loops = 0
+        # get ticket count
+    #    todo cursor.execute("SELECT * from tickets where (# enter conditions))
+		ticket_set = cursor.fetchall()
+
+        # Show 5 tickets if more than 5 and allow user to see more
+        if count == 5 or ticket == ticket_set[len(ticket_set) - 1]:
+            choice = input("Select one of these tickets? (Enter option # or press enter to see more)")
+            if choice == '':
+                count = -1
+                loops +=1
+            else:
+                try:
+                    choice = int(choice)
+                    return ticket_set[(loops * 4) + (choice - 1)][0]
+                except ValueError:
+                    print("Invalid option. Please retry.")
+                    # return 
+        count += 1
+
+
+    def issue(self, args):
+        rno = args[1]
+        fname = args[2]
+        lname = args[3]
+        make = args[4]
+        model = args[5]
+        year = args[6]
+        color = args[7]
+        violation = args[8]
+        fine = args[9]
+        vdate = str(dt.now())
+        tno = randint(0,999999)
```

Comment:
This is what I am referring too, we should change this to use a uuid library so that we avoid duplicate ids, which I am sure the TAs will dock us for